src/Shared/CommunicationsUtilities.cs

```diff
@@ -925,19 +924,23 @@ internal static HandshakeOptions GetHandshakeOptions(
             {
                 context |= HandshakeOptions.NodeReuse;
             }
+
             if (lowPriority)
             {
                 context |= HandshakeOptions.LowPriority;
             }
-#if FEATURE_SECURITY_PRINCIPAL_WINDOWS
+
+#if FEATURE_SECURITY_PRINCIPAL_WINDOWS || (RUNTIME_TYPE_NETCORE && ISWINDOWS)
```

Comment:
Don't use the compile-time `ISWINDOWS` check here; use a run-time check (so we can build the core MSBuild/SDK on any OS and have it work on any OS).